rpmio/rpmpgp.c

```diff
@@ -556,10 +556,8 @@ static int pgpPrtSigParams(pgpTag tag, uint8_t pubkey_algo, uint8_t sigtype,
 	int mpil = pgpMpiLen(p);
 	if (pend - p < mpil)
 	    break;
-	if (sigtype == PGPSIGTYPE_BINARY || sigtype == PGPSIGTYPE_TEXT) {
-	    if (sigalg->setmpi(sigalg, i, p))
-		break;
-	}
+	if (sigalg->setmpi(sigalg, i, p))
+	    break;
```

Comment:
This requires a corresponding change in the package signature checking code to ensure that package signatures are `PGPSIGTYPE_BINARY`.  #1705 is one implementation, and I can replace it with a better one that uses proper accessor functions.

Reply:
Yes, you've repeated this quite a few times now.

The signature type information is there to tell the reader how to has the material for correct results. We ignore the byte *anyhow* for the package hashing purposes because it's just not that intersting for our purposes.

A better implementation would do things differently in many ways, but removing that semi-random check from 20 years ago is hardly a security regression.

Reply:
@pmatilai good point.  In fact, I would argue that *not* adding the check would make this PR a regression security wise.  Would it be possible to include #1705 in this PR?  It is tiny and passes the regression test suite, and can be replaced with a better solution later.